pfCNI

```diff
@@ -1,108 +1,108 @@
 #!/bin/bash
 
-exec 2>> /var/log/pfcni.log
-
-network_conf=$(cat /dev/stdin)
-
-$gateway=192.168.20.1
-
-#IP ADDR for the pod
-IP_ADDR=192.168.20.10/24
-
-# Pick a namespace name
-mynetns=default-ns
-IFS=';'
-for i in $CNI_ARGS; do
-    echo $i >>  /var/log/demolog
-    if [[ $(echo $i | awk -F '=' '{print$1}') == K8S_POD_NAME ]]; then
-      mynetns=$(echo $i | awk -F '=' '{print$2}')
-      echo "netns: $mynetns" >>  /var/log/demolog
-    fi
-done
-
-func addVethPair{
-  mkdir -p /var/run/netns/
-  ln -sfT $CNI_NETNS /var/run/netns/$namespace
-  ip link add veth1 type veth peer name veth2
-  # Connect to some bridge if needed
-  # ip link set veth1 master <bridge>
-  ip link set dev veth1 up
-  ip link set veth2 netns $namespace
-  ip netns exec $namespace ip link set veth2 down
-  ip netns exec $namespace ip link set veth2 name $CNI_IFNAME
-  ip netns exec $namespace ip link set $CNI_IFNAME up
-  ip netns exec $namespace ip addr add $IP_ADDR dev $CNI_IFNAME
-  # Add route if needed
-  #ip netns exec $namespace ip route add default via .... dev $CNI_IFNAME
+readonly IP=169.255.0.0
+readonly SUBNET_MASK=24
+readonly GW_IP=169.255.0.1
+readonly CONTAINER_NETNS_LINK="/var/run/netns/$CNI_CONTAINERID"
+readonly LOGFILE=/var/log/pfcni.log
+
+function main() {
+    case "$CNI_COMMAND" in
+        ADD)
+            log "CNI_ARGS: $CNI_ARGS"
+            eval "$CNI_ARGS"
```

Comment:
`eval` frightens, it allows your caller to do dirty tricks.
I usually prefer parsing my input, rather than trusting it blindly.